### ISSUE TITLE
Export model without restart R session

### ISSUE THREAD

**Issue by quoctuams** [rstudio/keras#267]
Hi guys, I'm facing a quite weird situation as follows:
First, I ran a basic model like this:

```
model <- keras_model_sequential()
model %>%
    layer_dense(units = 256, activation = 'relu', input_shape = c(length(featureList))) %>%
    layer_dropout(rate = 0.4) %>%
    layer_dense(units = 128, activation = 'relu') %>%
    layer_dropout(rate = 0.3) %>%
    layer_dense(units = numCategory, activation = 'softmax')

model %>% compile(
    loss = 'categorical_crossentropy',
    optimizer = optimizer_rmsprop(),
    metrics = c('accuracy')
)
# train model
history <- model %>% fit(
    x_train, y_train,
    epochs = 30, batch_size = 25,
    validation_split = 0.2
)
```

The accuracy was about 60%.
After that, I exported the model using export_savedmodel and re-run the above code (without restarting R as suggested in the console). Surprisingly the accuracy now reached 90% (on training) and 80% (on validating).
This situation happens again & again with different sets of data. Do you guys have any idea what is going on?

**Comment by jjallaire** [comment]
If you are continuing to train the same model then it's certainly possible that the accuracy keeps improving. The best way for us to help you with this is if you provide a complete reproducible examples (i.e. in the example above there isn't data nor the code you use to export).

**Comment by quoctuams** [comment]
Thank you @jjallaire . Below is the the code and data:
```
library(keras)
x_train = as.matrix(read.csv("gh.x_train.csv"))
y_train = as.matrix(read.csv("gh.y_train.csv"))


# FIRST RUN BEFORE EXPORTING THE MODEL
model <- keras_model_sequential()
model %>%
    layer_dense(units = 256, activation = 'relu', input_shape = c(ncol(x_train))) %>%
    layer_dropout(rate = 0.4) %>%
    layer_dense(units = 128, activation = 'relu') %>%
    layer_dropout(rate = 0.3) %>%
    layer_dense(units = 3, activation = 'softmax')

model %>% compile(
    loss = 'categorical_crossentropy',
    optimizer = optimizer_rmsprop(),
    metrics = c('accuracy')
)

# train model
history <- model %>% fit(
    x_train, y_train,
    epochs = 30, batch_size = 25,
    validation_split = 0.2
)

# Results: loss: 0.8018 - acc: 0.6580 - val_loss: 0.8984 - val_acc: 0.6130



# ==========
export_savedmodel(model, "model-test")
# ==========



# SECOND RUN AFTER EXPORTING THE MODEL
model <- keras_model_sequential()
model %>%
    layer_dense(units = 256, activation = 'relu', input_shape = c(ncol(x_train))) %>%
    layer_dropout(rate = 0.4) %>%
    layer_dense(units = 128, activation = 'relu') %>%
    layer_dropout(rate = 0.3) %>%
    layer_dense(units = 3, activation = 'softmax')

model %>% compile(
    loss = 'categorical_crossentropy',
    optimizer = optimizer_rmsprop(),
    metrics = c('accuracy')
)

# train model
history <- model %>% fit(
    x_train, y_train,
    epochs = 30, batch_size = 25,
    validation_split = 0.2
)
# Results: loss: 0.5042 - acc: 0.7992 - val_loss: 1.0149 - val_acc: 0.6140
```
My understanding is that when I run the code
`model <- keras_model_sequential()`
the model already resets to blank? So why do 2 runs (before & after exporting the model; same data) produce quite different results. With some other datasets, the validation accuracy of the second run can get up to 90%.

Please see if you can reproduce this situation. Thank you very much!

[gh.train.data.zip](https://github.com/rstudio/keras/files/1686136/gh.train.data.zip)

**Comment by jjallaire** [comment]
Here it looks like your validation accuracy is identical on the second run?

In the most recent version of keras we print a warning saying that you should restart your R session after calling export_savedmodel (since that function manipulates the keras global learning phase). Are you seeing this warning?

**Comment by jjallaire** [comment]
I'm not able to reproduce this behavior (I see the validation accuracy stay around 61%, the same as in the output you shared above)

**Comment by quoctuams** [comment]
Yes I did see the warning, but one time I forgot to restart R session and what happened next was quite strange :)
In the example above, I cut the training dataset to only 5,000 rows so that may explain why the validation accuracy didnt change. Did you also see the same training accuracy?
Would you mind re-run the code with the new data I just attached in this comment? Thank you very much @jjallaire 
[gh.data.zip](https://github.com/rstudio/keras/files/1686311/gh.data.zip)

**Comment by jjallaire** [comment]
What's happening is that with the learning phase set to 0, the dropout layers are effectively disabled, so your model ends up with much worse overfitting (thus the high delta between accuracy and validation accuracy). You'll see the same behavior in a fresh session if you remove the dropout layers:

```r
model <- keras_model_sequential()
model %>%
  layer_dense(units = 256, activation = 'relu', input_shape = c(ncol(x_train))) %>%
  layer_dense(units = 128, activation = 'relu') %>%
  layer_dense(units = 3, activation = 'softmax')
```

**Comment by quoctuams** [comment]
Oh I see 👍 Thank you @jjallaire , appreciate it!
